File: app/src/main/java/cf/movie/slmovie/utils/rnUtils/download/DownloadModule.java
package cf.movie.slmovie.utils.rnUtils.download;

import android.content.Intent;
import android.net.Uri;

import com.facebook.react.bridge.Arguments;
import com.facebook.react.bridge.Promise;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactContextBaseJavaModule;
import com.facebook.react.bridge.ReactMethod;
import com.facebook.react.bridge.WritableMap;

/**
 * Created by 包俊 on 2017/8/9.
 */

public class DownloadModule extends ReactContextBaseJavaModule {

    private ReactApplicationContext context;

    public DownloadModule(ReactApplicationContext reactContext) {
        super(reactContext);
        this.context = reactContext;
    }

    @Override
    public String getName() {
        return "DownloadNative";
    }

    @ReactMethod
    public void pushDownload(String url, Promise promise) {
        boolean isFind = true;
        try {
            Intent intent = new Intent(Intent.ACTION_VIEW, Uri.parse(url));
            intent.addCategory("android.intent.category.DEFAULT");
            context.startActivity(intent);
        } catch (Exception e) {
            isFind = false;
            e.printStackTrace();
        }
        WritableMap maps = Arguments.createMap();
        maps.putBoolean("result", isFind);
        promise.resolve(maps);
    }
}

File: app/src/main/java/cf/movie/slmovie/utils/rnUtils/download/DownloadModule.kt
package cf.movie.slmovie.utils.rnUtils.download

import android.content.ClipData
import android.content.ClipboardManager
import android.content.Context
import android.content.Intent
import android.net.Uri
import android.widget.Toast
import com.facebook.react.bridge.*

/**
 * Created by 包俊 on 2017/8/9.
 */

class DownloadModule(private val context: ReactApplicationContext) : ReactContextBaseJavaModule(context) {

    override fun getName(): String {
        return "DownloadNative"
    }

    @ReactMethod
    fun pushDownload(url: String, promise: Promise) {
        var isFind = true
        try {
            val intent = Intent(Intent.ACTION_VIEW, Uri.parse(url))
            intent.addCategory("android.intent.category.DEFAULT")
            context.startActivity(intent)
        } catch (e: Exception) {
            Toast.makeText(context, url, Toast.LENGTH_LONG).show()
            val myClipboard = context.getSystemService(Context.CLIPBOARD_SERVICE) as ClipboardManager
            val myClip = ClipData.newPlainText("text", url)
            myClipboard.primaryClip = myClip
            isFind = false
            e.printStackTrace()
        }

        val maps = Arguments.createMap()
        maps.putBoolean("result", isFind)
        promise.resolve(maps)
    }
}
